### ISSUE TITLE
Change default priority to match Apple?

### ISSUE THREAD

**Issue by BrianKenny** [jmfieldman/Mortar#9]
I'm curious what the thinking is on mortar constraints defaulting to a lower priority(500) than constraints created with Apple's methods(required), and if you'd consider changing that?  I had to add a bunch of ! .Requrieds until I found MortarDefault.Priority.setBase(.Required).  Just recently I recommended Mortar to a team of devs for a different app, and within a day or two got asked why mortar constraints don't work constraining image view sizes.  They do of course, but by default their priority is too low(500) to override the image's default compression resistance(750).  I would imagine many/most new users will be surprised by this, and some will probably just assume it's a bug & bail.

**Comment by jmfieldman** [comment]
Hi @BrianKenny -- sorry for the delay, for some reason I didn't get an email notifying me about this issue.

The basic summary is:  I created this library back when I was first doing Auto Layout stuff and found it to be super-annoying in code.  Tried SnapKit and didn't like blocks everywhere.  For whatever reason when I was looking into the priority aspect, I thought that 500 was normal/default, and that is what I made as the default for the first N versions of Mortar.  

So even in my own production code in the early versions of Mortar I was putting ! .required everywhere and was SUPER ANNOYED by it.  That's when I put in the whole default stack/setbase thing so I could initialize it to .required when the app launched -- the whole time being annoyed at Apple that required wasn't the default.

And then at some point I realized that .required WAS the default.  But now I'm faced with the migration issue of what to do about existing users.  I feel like I can't just blindly make the new default be .required, since that might break existing codebases that assume the default is 500.  For those apps, the simple solution would be to add a set(base: 500) to the initialization of the app, but I don't know how to effectively communicate that to anyone updating the library.

Bottom line: I agree that .required should be the default but not sure how to cleanly change it w/o potentially screwing early adopters.

**Comment by siemensikkema** [comment]
I'm not sure there's anything you can do beyond using a different minor (or even major?) version number along with good release documentation and a mention in the README.md. Early adopters are supposed to be the most forgiving, right? :)

**Comment by jmfieldman** [comment]
Yeah, there's basically no guaranteed hook I can message people through, so I should just bite the bullet and change it in a 1.1 release and hope I can call it out as well as possible in documentation.  I'll integrate that and your pull request when I get home later today.

**Comment by jmfieldman** [comment]
Closing ref: https://github.com/jmfieldman/Mortar/commit/632f2c004b6bd3578f99d24a03482b5f94eaefb0

**Comment by BrianKenny** [comment]
That's pretty much what I suspected.  Hindsight ehh  :)   I'm on board with the suggestion to change it & bump the version.  I suspect most everyone who's using Mortar in anger has already set the the default and won't even notice the change.
